components/tooltip/src/tooltip.test.js

```diff
@@ -187,10 +184,27 @@ describe('Keyboard interaction', () => {
 
         // open tooltip
         wrapper.simulate('mouseover')
+
         act(() => {
-            jest.advanceTimersByTime(200) // open the tooltip
+            jest.advanceTimersByTime(200)
         })
-        document.dispatchEvent(new KeyboardEvent('keydown', { key: 'Escape' }))
+
+        // verify tooltip is open
+        expect(document.querySelector(tooltipContentSelector)).not.toBe(null)
+
+        //Press the Escape key
+        act(() => {
+            document.dispatchEvent(
```

Comment:
are the `act`s around the timer stuff necessary? Also instead of `document.dispatch`, are you not able to use the `fireEvent` from react-testing-library itself? https://testing-library.com/docs/dom-testing-library/api-events/#fireevent